server.js

```
@@ -0,0 +1,180 @@
+const fs = require("fs");
+const querystring = require("querystring");
+const post = require("./post");
+const newHtml = require("./newHtml");
+const PORT = process.env.PORT || 8080;
+
+function findUrl(data) {
+  if (data === "/" || data === "/index.html" || data === "/favicon.ico") {
+    return "index.html";
+  } else {
+    return data;
+  }
+}
+
+const server = http.createServer((req, res) => {
+  let body = "";
+  req.on("data", data => (body += data));
+
+  req.on("end", () => {
+    let filePath = path.join(__dirname, "public", findUrl(req.url));
+    let extName = path.extname(filePath);
+    let contentType = "text/html";
+    if (extName === ".css") {
+      contentType = "text/css";
+    }
+
+    // GET Request
+    if (req.method === "GET") {
+      fs.readFile(filePath, "utf8", (err, content) => {
+        if (err) {
+          fs.readFile(
+            path.join(__dirname, "public", "404.html"),
+            "utf8",
+            (err, content) => {
+              if (err) throw err;
+              res.writeHead(200, {
+                "Content-Type": "text/html",
+                "Content-Length": content.length
+              });
+              res.write(content);
+              res.end();
+            }
+          );
+        } else {
+          res.writeHead(200, {
+            "Content-Type": contentType,
+            "Content-Length": content.length
+          });
+          res.write(content);
+          res.end();
+        }
+      });
+    }
+
+    // POST Request
+    if (req.method === "POST") {
+      if (req.url === "/elements") {
+        let q = querystring.parse(body);
+        let elName = q.elementName;
+        let elSymbol = q.elementSymbol;
+        let elNum = q.elementAtomicNumber;
+        let elDescript = q.elementDescription;
+
+        fs.writeFile(
+          path.join(__dirname, "/public", `${elName}.html`),
+          post(elName, elSymbol, elNum, elDescript),
+          err => {
+            if (err) throw err;
```

This should return an error to the client using either a 400/500 error